src/test/java/server/response/PathFinderTest.java

```diff
@@ -0,0 +1,48 @@
+package test.java.server.response;
+
+import main.java.server.request.Request;
+import main.java.server.request.StatusLine;
+import main.java.server.response.PathFinder;
+import main.java.server.response.Response;
+import org.junit.Before;
+import org.junit.Test;
+
+import static junit.framework.TestCase.assertEquals;
+
+public class PathFinderTest {
+    private PathFinder pathFinder;
+    private AppStub app;
+
+    @Before public void setUp() {
+        new RoutesStub();
+        app = new AppStub(RoutesStub.ROUTES);
+    }
+
+    @Test
+    public void findsMatchingRouteForSimpleGetRequestAndGeneratesResponse() {
+        Request simpleGetRequest = new Request(new StatusLine("GET", "/simple_get", "HTTP1.1"));
+        pathFinder = new PathFinder(simpleGetRequest, app);
+        Response simpleGetResponse = pathFinder.selectHandler();
+
+        assertEquals("HTTP/1.1 200 OK\r\n", simpleGetResponse.getStatusLine());
+        }
+
+    @Test
+    public void returnsNotFoundResponseForUnknownRoute() {
+        Request unknownRequest = new Request(new StatusLine("GET", "/WHO_GOES_THERE", "HTTP1.1"));
+        pathFinder = new PathFinder(unknownRequest, app);
+        Response unknownResponse = pathFinder.selectHandler();
+
+        assertEquals("HTTP/1.1 404 Not Found\r\n", unknownResponse.getStatusLine());
+    }
+
+    @Test
+    public void returnsMethodNotAllowedForRequestWithForbiddenMethod() {
+        Request wrongMethodRequest = new Request(new StatusLine("PUT", "/simple_get", "HTTP1.1"));
+        pathFinder = new PathFinder(wrongMethodRequest, app);
+        Response unknownResponse = pathFinder.selectHandler();
```

Comment:
wonder if there might be a name other than `unknownResponse` that wouldn't overlap with the name we're using for a 404 response above